api/api/specs/experiment-engines.yaml

```diff
@@ -270,23 +270,6 @@ components:
         variables:
           $ref: "#/components/schemas/ExperimentVariables"
 
-    StandardExperimentConfig:
-      type: "object"
-      properties:
-        client:
-          $ref: "#/components/schemas/ExperimentClient"
-        experiments:
-          type: array
-          items:
-            $ref: "#/components/schemas/Experiment"
-        variables:
-          $ref: "#/components/schemas/ExperimentVariables"
-
-    CustomExperimentConfig:
-      type: "object"
-      not:
-        $ref: "#/components/schemas/StandardExperimentConfig"
```

Comment:
I understand that you are removing this because of some limitations of https://github.com/OpenAPITools/openapi-generator, but this is not a viable solution to remove these schemas completely as they (especially `StandardExperimentConfig`) are helpful source of the API documentation. 
You need to explore other possible solutions to the limitation. Some existing parts of the SDK are already using `oneof` here: https://github.com/gojek/turing/blob/1f22856d7f5fd647ef3bbb75fe889ab73e29d1d0/api/api/specs/ensemblers.yaml#L131-L138
(however I'm not sure if it matches the issue you observed)

Reply:
Hmm, okay I made that change thinking that we'll not be explicitly enforcing specific values of `ExperimentEngineType` using an enum. I was previously planning to turn the original `ExperimentConfig` schema into a 'base class' and allow the derived schemas `StandardExperimentConfig` and `CustomExperimentConfig` to inherit from it directly, while using a discriminator on the `ExperimentEngineType` to determine which child schema to use:

```
StandardExperimentConfig:
  allOf:
    - $ref: "#/components/schemas/ExperimentConfig"
    - type: "object"
      properties:
        config:
          type: "object"
          properties:
            project_id:
              type: "integer"
            client:
              $ref: "#/components/schemas/ExperimentClient"
            experiments:
              type: array
              items:
                $ref: "#/components/schemas/Experiment"
            variables:
              $ref: "#/components/schemas/ExperimentVariables"

CustomExperimentConfig:
  allOf:
    - $ref: "#/components/schemas/ExperimentConfig"
    - type: "object"

ExperimentConfig:
  type: "object"
  required:
    - type
  properties:
    type:
      $ref: "#/components/schemas/ExperimentEngineType"
  discriminator:
    propertyName: type
    mapping:
      'nop': "#/components/schemas/StandardExperimentConfig"
      'xp': "#/components/schemas/CustomExperimentConfig"
      'litmus': "#/components/schemas/StandardExperimentConfig"

# Enums
ExperimentEngineType:
  type: "string"
  default: "nop"
```

Would this be a more viable option?

-----

I think the example on the `Ensembler` using `oneOf` didn't encounter any issues with the OpenAPI generator because it only contained 1 schema to choose from (it inherently behaves as an `allOf` in that case), unlike the original specification of `ExperimentConfig`.

Reply:
> Would this be a more viable option?

Actually, no, because the list of supported exp engines is dynamic and can vary from one deployment to another. It seems like the support of `oneof` is [finally coming](https://github.com/OpenAPITools/openapi-generator/issues/10812) to openapi generator, so I think for now we can change `ExperimentConfig` to 
```yaml
ExperimentConfig:
      type: "object"
      required:
        - type
      properties:
        type:
          $ref: "#/components/schemas/ExperimentEngineType"
        config:
          type: "object"
```
But at the same time, we can keep `#/components/schemas/StandardExperimentConfig` and  `#/components/schemas/CustomExperimentConfig` definitions in this spec so when `oneof` keyword is supported we can revert it back.